### PR TITLE
feat(ST-50): endpoint for saving message on database

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change

- Created Module, controller and service layer for messages
- Implement endpoint for saving messages on the database
- Message.serializer is used to shape the api response
- Imported the Message module in app module

## Associated Ticket Link(s)

- https://trello.com/c/YOoTinqz

## Screenshots / Screen Recordings

N/A

### CODE REVIEW

File: src/messages/message.controller.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    private readonly messagesService: MessagesService,
+    private readonly messageSerializer: MessagesSerializer,
+  ) {}
+
+  @Post()
+  async createMessage(
+    @CurrentUser() user: { id: number },
+    @Body() createMessageDto: CreateMessageDto,
+  ) {
+    const message: Message = await this.messagesService.createMessage(createMessageDto, user.id);
```
Comment (ishmam-iqbal-sazim) [inline src/messages/message.controller.ts]: Let's add new line below

File: src/messages/messages.repository.ts [hunk 2]
```diff
@@ -4,5 +4,13 @@ import { EntityRepository } from "@mikro-orm/postgresql";
 
 import { Message } from "@/common/entities/messages.entity";
 
+import { ICreateMessageData } from "./message.interfaces";
+
 @Injectable()
-export class MessageRepository extends EntityRepository<Message> {}
+export class MessageRepository extends EntityRepository<Message> {
```
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.repository.ts]:
```suggestion
export class MessagesRepository extends EntityRepository<Message> {
```
[NIT]

File: src/messages/message.controller.ts
Comment (ishmam-iqbal-sazim) [inline src/messages/message.controller.ts:1]: Entity in filenames should be plural i.e. messages.controller, messages.service, messages.dtos etc.

File: src/messages/messages.repository.ts (hunk 2 above)
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.repository.ts]: Let's add new line below

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending nits

File: src/messages/messages.controller.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+@UseGuards(JwtAuthGuard)
+@Controller("messages")
+export class MessagesController {
+  constructor(
+    private readonly messagesService: MessagesService,
+    private readonly messagesSerializer: MessagesSerializer,
+  ) {}
+
+  @Post()
```
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.controller.ts:19]: Endpoint should be `classrooms/:classroomId/messages` as per REST principles.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**